util/httputil/http_util.go

```diff
@@ -67,7 +65,19 @@ func newAuthRoundTripper(auth *config.Auth, rt http.RoundTripper) http.RoundTrip
 	}
 }
 
-func AuthTransport(auth *config.Auth, transportConfig *http.Transport) (http.RoundTripper, error) {
+func CreateTransport(auth *config.Auth, transportConfig *http.Transport, timeout time.Duration) (http.RoundTripper, error) {
+	// Limits the time spent establishing a TCP connection if a new one is
+	// needed.
+	transportConfig.Dial = (&net.Dialer{
+		Timeout:   timeout,
+		KeepAlive: timeout,
+	}).Dial
+	transportConfig.IdleConnTimeout = timeout
```

Comment:
Unless the httpClient is reused, I don't think the Idle Timeout should match the Dial Timeout.

I think this is more to enable reusing connections and have a small connection pool. But in Kiali, I think we create a new client every time. So, I think it's better to keep this to a static low value that can be overridden via an optional argument.

This is just a suggestion. If you don't agree, I think it's OK to keep as is.

Reply:
> But in Kiali, I think we create a new client every time. So, I think it's better to keep this to a static low value that can be overridden via an optional argument.

True except precisely for the Prometheus client case, which is reused. I guess this is the main takeaway for this issue: have the idle timeout set for prometheus.

Reply:
In the graph, the prometheus client is bound to the request lifetime. Same for a couple of other handlers.
But in parts that use the "business" layer, it's different, see here: https://github.com/kiali/kiali/blob/master/business/layer.go#L87-L99
The prometheus client is global hence re-used.
The jaeger client is created via a loader, hence it is only created on demand.

Without even questioning if it's better or not to re-use a client, I'd say it is safer to define the idle timeout anyway, like @cfcosta did, to be more future-proof.

Reply:
My concern was that if you set a large timeout (say, for example, 2 minutes), when seeing this diagram: https://blog.cloudflare.com/content/images/2016/06/Timeouts-002.png.

what I understand is that the client will wait up to 2 minutes for the request and response to complete. Once that's done, the connection becomes idle and there is another 2 minute wait giving the connection a maximum of 4 minutes of lifespan. This is OK if the connection is going to be reused (which is the case for Prometheus client in Kiali). But if it's not going the be re-used, the connection is allowed to be there idle for 2 minutes.

For the scope of this PR, keeping as is this is quite fine, because we are using short timeouts. So, I'm good and as @jotak said, this is better than nothing. But I would like to, at least, see a comment in the function so that we don't forget that timeouts should be "reasonably short" to prevent any hanging connection in idle state.

Reply:
@israel-hdez I agree with your points, added a comment explaining the problem so people in the future can be aware.

Reply:
> But if it's not going the be re-used (which is the case for non-p8s connections), the connection is allowed to be there idle for 2 minutes.

I think (hope) it is not the case, when the client is not reused, it is closed, hence any connections that would be idle would be closed immediately as well, I guess